### ISSUE TITLE
`gmem` can cause segfault

### ISSUE THREAD

**Issue by hughperkins** [jrprice/Oclgrind#104]
```
(oclgrind) lmem

   1000000000000: 00 00 00 00
   1000000000004: 00 00 00 00
   1000000000008: 00 00 00 00
   100000000000C: 90 1B 51 3F
   1000000000010: 00 00 00 00
   1000000000014: 00 00 00 00
   1000000000018: 00 00 00 00
   100000000001C: 00 00 00 00
   2000000000000: 00 00 00 00
   2000000000004: 00 00 00 00
   2000000000008: 00 00 00 00
   200000000000C: 00 00 80 40
   2000000000010: 00 00 00 00
   2000000000014: 00 00 00 00
   2000000000018: 00 00 00 00
   200000000001C: 00 00 00 00
   3000000000000: 00 00 00 00
   3000000000004: 00 00 00 00
   3000000000008: 00 00 00 00
   300000000000C: 01 00 00 00
   3000000000010: 00 00 00 00
   3000000000014: 00 00 00 00
   3000000000018: 00 00 00 00
   300000000001C: 00 00 00 00
(oclgrind) gmem

   1000000000000: 00 00 00 00
   1000000000004: 00 00 00 00
   1000000000008: 00 00 00 00
   100000000000C: 00 00 00 00
   2000000000000: 34 29 6F 3F
   2000000000004: CA F2 18 3D
   2000000000008: 2E 8D 37 3E
   200000000000C: 90 1B 51 3F
   2000000000010: 60 C4 2E 3B
   3000000000000: 34 29 6F 3F
   3000000000004: CA F2 18 3D
   3000000000008: 2E 8D 37 3E
   300000000000C: 90 1B 51 3F
   3000000000010: 60 C4 2E 3B
   4000000000000: 00 00 80 3F
   4000000000004: 00 00 00 40
   4000000000008: 00 00 40 40
   400000000000C: 00 00 80 40
Segmentation fault
```

**Comment by jrprice** [comment]
I can't reproduce this, can you provide a simple test case that exhibits the issue?

**Comment by jrprice** [comment]
Finally reproduced this, should be fixed in 8d6955c.

**Comment by hughperkins** [comment]
Great!  Thanks!
